app/src/main/java/com/tyron/code/ui/iconmanager/adapter/IconAdapter.java

```diff
@@ -0,0 +1,132 @@
+package com.tyron.code.ui.iconmanager.adapter;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.PorterDuff;
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import com.tyron.code.ui.iconmanager.IconManagerActivity;
+import com.tyron.code.R;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class IconAdapter extends RecyclerView.Adapter<IconAdapter.ViewHolder> {
+
+    ArrayList<String> data;
```

Comment:
use an ArrayList of `java.io.File` instead to remove unecessary conversions to file again